x-pack/plugins/ml/public/file_datavisualizer/components/import_view/importer/importer_factory.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { CsvImporter } from './csv_importer';
+import { SstImporter } from './sst_importer';
+import { JsonImporter } from './json_importer';
+
+export function importerFactory(format, results) {
+
+  let importer;
+  switch (format) {
+    case 'delimited':
+      importer = new CsvImporter(results);
+      break;
+    case 'semi_structured_text':
+      importer = new SstImporter(results);
+      break;
+    case 'json':
+      importer = new JsonImporter(results);
+      break;
+    default:
+      break;
```

Comment:
maybe worth throwing an error if the code asks for an unsupported imported - or are we intentionally failing silently?

Reply:
The format that is supported by the backend but not the UI is XML.  I agree it would be nice to show an error like "XML not currently supported" if this is detected.  But would this already have been done at an earlier stage, meaning that the import button wouldn't even be available in this case?  If so then some sort of assertion that the default branch should be impossible would be OK here.

Reply:
I will make a change to flag xml as an unsupported format when the user first analyses the file.
They will then not get near this import stage.